pkg/api/server/v1alpha2/plugin/plugin_logs.go

```diff
@@ -250,8 +271,133 @@ func (s *LogPluginServer) getLokiLogs(writer io.Writer, parent string, rec *db.R
 
 }
 
+func getBlobLogs(s *LogServer, writer io.Writer, parent string, rec *db.Record) error {
+	URL, err := url.Parse(s.config.LOGGING_PLUGIN_API_URL)
+	if err != nil {
+		s.logger.Error(err)
+		return err
+	}
+
+	legacy := false
+	queryParams := URL.Query()
+
+	for k, v := range s.queryParams {
+		if k == legacyLogType && v == "true" {
+			legacy = true
+			continue
+		}
+		queryParams.Add(k, v)
+	}
+
+	logPath := map[string]string{}
+
+	ctx := context.Background()
+	bucket, err := openBucket(ctx, URL.String()+"?"+queryParams.Encode())
+	if err != nil {
+		s.logger.Errorf("error opening bucket: %s", err)
+		return err
+	}
+	defer clean(bucket, s.logger)
+
+	switch rec.Type {
+	case typePipelineRun:
+		err := errors.New("pipelinerun not supported, please use taskrun")
+		s.logger.Error(err.Error())
+		return err
+	case typeTaskRun:
+		logRec := &db.Record{}
+		if legacy {
+			logRec, err = getLogRecord(s.db, parent, rec.ResultID, rec.Name)
+			if err != nil {
+				s.logger.Debugf("error getting legacy log record: %s", err)
```

Comment:
why do we continue here?
1. If err != nil, then logRec will be nil
2. If legacy is configured which means log type to v1alpha2, why do we continue as v1alpha3?

Reply:
1. Yes, fixed that.
2. Yes, Log Type kind itself can be v1alpha2 or  v1alpha3. 
In plugin API, there's no LogType record itself.